Core/Core/Assignments/AssignmentList/ViewModel/AssignmentCellViewModel.swift

```diff
@@ -21,10 +21,10 @@ import SwiftUI
 public class AssignmentCellViewModel: ObservableObject {
     public let assignment: Assignment
     public let courseColor: UIColor?
+    private let env: AppEnvironment
 
-    private let env = AppEnvironment.shared
-
-    public init(assignment: Assignment, courseColor: UIColor?) {
+    public init(env: AppEnvironment, assignment: Assignment, courseColor: UIColor?) {
```

Comment:
I think we usually pass in the primary data/model first which is the main subject of the whole type.
Then the secondary ones.
And in the end we pass in the dependecies, like interactors, env, router, etc.
Maybe this is only me, but it would make sense to me to keep this ordering.
May look like nit, but I would rather call this out now, at the beginning of this approach.

The same stands for all the other places where we inject `env` in this PR. cc @vargaat for thoughts

Reply:
Yes, I agree, that bloated environment object is definitely not a primary dependency. I didn't notice this during my review but let's keep the priority order in the future.

Reply:
Yes, that's right.
If possible, I will have a follow up PR after release to address that.